File: src/Ilc.Data.Models/Trainer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ilc.Data.Models.Base;

namespace Ilc.Data.Models
{
    public class Trainer : CompanyBaseEntity
    {
        public string Name { get; set; }
        public string Phone { get; set; }
        public string Address { get; set; }

        public virtual ICollection<Subject> Subjects { get; set; }

        public int UserProfileId { get; set; }
        [ForeignKey("UserProfileId")]
        public virtual UserProfile UserProfile { get; set; }   
    }
}

File: src/Ilc.Web/Scripts/app/model/Trainer.js
Ext.define('Ilc.model.Trainer', {
    extend: 'Ext.data.Model',
    
    fields: [
        'id',
        {
            name: 'name'
        },
        {
            name: 'email'
        },
        'userInfo',
        'subjects'
    ]
});